qCC/ccCommandLineParser.cpp

```diff
@@ -244,12 +245,34 @@ QString ccCommandLineParser::exportEntity(	CLEntityDesc& entityDesc,
 		return "[ExportEntity] Internal error: invalid input entity!";
 	}
 
+	bool anyForced = options.testFlag(ExportOption::ForceCloud) | options.testFlag(ExportOption::ForceHierarchy) | options.testFlag(ExportOption::ForceMesh);
 	//specific case: clouds
-	bool isCloud = entity->isA(CC_TYPES::POINT_CLOUD);
-	isCloud |= forceIsCloud;
-	QString extension = isCloud ? m_cloudExportExt : m_meshExportExt;
+	bool isCloud = entity->isA(CC_TYPES::POINT_CLOUD) || entityDesc.getCLEntityType() == CL_ENTITY_TYPE::CLOUD;
 
-	QString outputFilename = getExportFilename(entityDesc, extension, suffix, baseOutputFilename, forceNoTimestamp);
+	//specific case: mesh
+	bool isMesh = entity->isKindOf(CC_TYPES::MESH) || entityDesc.getCLEntityType() == CL_ENTITY_TYPE::MESH;
+
+	QString extension = isCloud ? m_cloudExportExt : isMesh ? m_meshExportExt : m_hierarchyExportExt;
```

Comment:
Here shouldn't we test if the entity is a hierarchy?

Reply:
well with only three options I didn't test for hierarchies, if isCloud or isMesh else hierarchy.
I did set up a bool isHierarchy but removed it when I realized I didn't need it. If we have add another category it could be added.

Reply:
Ah sorry, I read too fast (I was confused by the cascading '?' tests.